pkg/kubectl/cmd/replace.go

```diff
@@ -154,13 +223,12 @@ func RunReplace(f cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		}
 
 		info.Refresh(obj, true)
-		cmdutil.PrintSuccess(shortOutput, out, info.Object, false, "replaced")
-		return nil
+		return o.PrintObj(info.AsVersioned())
 	})
 }
 
-func forceReplace(f cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string, shortOutput bool, options *resource.FilenameOptions) error {
-	schema, err := f.Validator(cmdutil.GetFlagBool(cmd, "validate"))
+func (o *ReplaceOpts) forceReplace(f cmdutil.Factory, args []string) error {
+	schema, err := f.Validator(o.validate)
```

Comment:
Why do you care about the `Validator` call if you already did that before getting here?

Reply:
We are obtaining a schema from it to use in the builder. I have moved this to a field in the options struct